test/parallel/test-find-package-json.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+const common = require('../common');
+const fixtures = require('../common/fixtures');
+const tmpdir = require('../common/tmpdir');
+const assert = require('node:assert');
+const fs = require('node:fs');
+const { findPackageJSON } = require('node:module');
+const { describe, it } = require('node:test');
+const { pathToFileURL } = require('node:url');
+
+
+describe('findPackageJSON', () => { // Throws when no arguments are provided
+  it('should throw when no arguments are provided', () => {
+    assert.throws(
+      () => findPackageJSON(),
+      { code: 'ERR_INVALID_ARG_TYPE' }
+    );
+  });
+
+  it('should throw when parentURL is invalid', () => {
+    for (const invalid of ['', null, undefined, {}, [], Symbol(), 1, 0]) assert.throws(
+      () => findPackageJSON('', invalid),
+      { code: 'ERR_INVALID_ARG_TYPE' },
+      invalid
+    );
+  });
+
+  it('should accept a file URL (string), like from `import.meta.resolve()`', () => {
+    const importMetaUrl = `${pathToFileURL(fixtures.fixturesDir)}/whatever.ext`;
+    const specifierBase = './packages/root-types-field/';
+    assert.strictEqual(
+      findPackageJSON(`${specifierBase}index.js`, importMetaUrl),
+      fixtures.path(specifierBase, 'package.json')
+    );
+  });
+
+  it('should be able to crawl up (CJS)', () => {
+    const pathToMod = fixtures.path('packages/nested/sub-pkg-cjs/index.js');
+    const parentPkg = require(pathToMod);
+
+    const pathToParent = fixtures.path('packages/nested/package.json');
+    assert.strictEqual(parentPkg, pathToParent);
+  });
+
+  it('should be able to crawl up (ESM)', () => {
+    const pathToMod = fixtures.path('packages/nested/sub-pkg-esm/index.js');
+    const parentPkg = require(pathToMod).default; // This test is a CJS file
+
+    const pathToParent = fixtures.path('packages/nested/package.json');
+    assert.strictEqual(parentPkg, pathToParent);
+  });
+
+  it('can require via package.json', () => {
+    const pathToMod = fixtures.path('packages/cjs-main-no-index/other.js');
+    // `require()` falls back to package.json values like "main" to resolve when there is no index
+    const answer = require(pathToMod);
+
+    assert.strictEqual(answer, 43);
+  });
+
+  it('should be able to do whatever the heck Antoine created', () => {
```

Comment:
@aduh95 can you think of a good description of what this is doing?

Reply:
```suggestion
  it('should be able to resolve both root and closest package.json', () => {
```